update.sh: avoid mutation of the running script

ArneBab opened the pull request.

Commit:
update.sh: avoid mutation of the running script

xor-freenet commented on scripts/update.sh:
```diff
@@ -255,6 +255,17 @@ then
+}
+trap recover_update_sh EXIT
+# rename the current script to ensure that we do not override what we are executing
+mv $0 update_tmp.sh && cp update_tmp.sh update.sh
```
I don't know precisely how /bin/sh, which this script is configured to run as, behaves compared to /bin/bash, but with bash this would have to be:
```mv "$0" ...```
See http://mywiki.wooledge.org/BashPitfalls#cp_.24file_.24target

Further, independent of the shell as that is a feature of mv, this actually needs to be:
```mv -- ...```
See http://mywiki.wooledge.org/BashPitfalls#Filenames_with_leading_dashes

ArneBab replied:
ah, yes, thank you!

Commit:
avoid argument quoting and option pitfalls

Thanks to xor

Commit:
update.sh: recovery works with different name

Commit:
update.sh: FIX comment

the wrapper around updating update.sh only ensure
recovery on exit during update. Mutation is stopped
by the early mv.

Commit:
update.sh: can again update freenet-stable-latest.jar

This is very, very basic functionality: partial duct tape
to compensate for the loss of our downloads infrastructure.

It is the bare minimum needed to be able to release again.

xor-freenet commented:
Once you have merged code to fred which also deploys this to existing installations please mark this as resolved: https://bugs.freenetproject.org/view.php?id=6707

ArneBab commented:
@xor-freenet resolved — thanks!
